### ISSUE TITLE
Nan Values after stacking multiple layers

### ISSUE THREAD

**Issue by brennanaba** [lucidrains/egnn-pytorch#9]
Hi Lucid!!

I find that when stacking multiple layers the output from the model rapidly goes to Nan. I suspect it may be related to the weights used for initialization.

Here is a minimal working example:
  
## Make some data: 
        import numpy as np
        import torch
        from egnn_pytorch import EGNN
        
        torch.set_default_dtype(torch.double)

        zline = np.arange(0, 2, 0.05)
        xline = np.sin(zline * 2 * np.pi) 
        yline = np.cos(zline * 2 * np.pi)
        points = np.array([xline, yline, zline])
        geom = torch.tensor(points.transpose())[None,:]
        feat = torch.randint(0, 20, (1, geom.shape[1],1))

## Make a model:

        class ResEGNN(torch.nn.Module):
            def __init__(self, depth = 2, dims_in = 1):
                super().__init__()
                self.layers = torch.nn.ModuleList([EGNN(dim = dims_in) for i in range(depth)])
            
            def forward(self, geom, feat):
                for layer in self.layers:
                    feat, geom = layer(feat, geom)
                return geom

## Run model for varying depths:

        for i in range(10):
            model = ResEGNN(depth = i)
            pred = model(geom, feat)
            mean_absolute_value  = torch.abs(pred).mean()
            print("Order of predictions {:.2f}".format(np.log(mean_absolute_value.detach().numpy())))

Output :
Order of predictions -0.29
Order of predictions 0.05
Order of predictions 6.65
Order of predictions 21.38
Order of predictions 78.25
Order of predictions 302.71
Order of predictions 277.38
Order of predictions nan
Order of predictions nan
Order of predictions nan

**Comment by lucidrains** [comment]
@brennanaba Hey Brennan! Do you want to give 0.0.18 a try? I initialized the last linear layer in the coordinates MLP to be 0

**Comment by brennanaba** [comment]
Hi Phil!! Works amazing. Thanks again for the quick fix!!

**Comment by denjots** [comment]
Also struggling with the lack of stability of this network layer on larger sized tasks. I posted an issue before, thought I'd solved it, but then obviously hadn't, so closed the ticket. This patch certainly helps with the stability, but in my last run on graphs of < 100 nodes, after a reasonably good start of 58 epochs, the loss suddenly started exploding to NaN. I think the algorithm itself is fundamentally flawed - or perhaps not yet properly described. For small problems, it's no doubt fine. For larger graphs of differing sizes, I just can't see how the notion of calculating a weighted **sum** of coordinates can be stable. Maybe they mean't a weighted average rather than sum or maybe the relative coordinates should be normed to unit length first.
